src/main/kotlin/snyk/container/YAMLImageExtractor.kt

```diff
@@ -5,10 +5,25 @@ import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.psi.PsiFile
 import io.snyk.plugin.findPsiFileIgnoringExceptions
+import org.intellij.lang.annotations.Language
 
 object YAMLImageExtractor {
     private val logger = Logger.getInstance(javaClass)
-    private val imageRegEx = "(?<=([-\\s]image:))(\\s*[a-zA-Z0-9./\\-_]+)(:)?([a-zA-Z0-9./\\-_]+)?".toRegex()
+
+    // see https://kubernetes.io/docs/concepts/containers/images/
+    @Language("regexp")
+    private const val imageNameSymbol = "[a-zA-Z0-9./\\-_]"
+
+    @Language("regexp")
+    private const val imageName = "($imageNameSymbol+(:)?($imageNameSymbol+)?)"
+
+    @Language("regexp")
+    private const val imageNameDescription = "$imageName|(\"$imageName\")"
+
+    @Language("regexp")
+    private const val imageTag = "[-\\s]image:\\s*"
```

Comment:
It would be nice to get rid of the look-back in the regexp, but does this cover leading whitespace, comments and trailing comments in the line, e.g. 
```
#      - image: nxginx/basti:basti-latest-12
      - image: nxginx/basti:basti-latest-12 # trailing comment
```

Reply:
I'll try some yamls later to check :)

Reply:
tested with 
![image](https://user-images.githubusercontent.com/14268320/158612512-92ab892c-5ba3-4690-b12a-ca62917dc1e0.png)
only `debian` image used for container scan

Reply:
in the following PR I'm also going to handle cases for image's digest `@` and both port and tag existence inside the name: `fictional.registry.example:<port>/imagename:<tag>`